tools/build-kani/src/sysroot.rs

```diff
@@ -111,96 +111,119 @@ pub fn build_lib() {
         .spawn()
         .expect("Failed to run `cargo build`.");
 
-    // Remove kani "std" library leftover.
-    filter_kani_std(&mut cmd);
+    // Collect the build artifacts.
+    let artifacts = build_artifacts(&mut cmd);
     let _ = cmd.wait().expect("Couldn't get cargo's exit status");
 
-    // Create sysroot folder.
+    // Create sysroot folder hierarchy.
     let sysroot_lib = kani_sysroot_lib();
     sysroot_lib.exists().then(|| fs::remove_dir_all(&sysroot_lib));
-    fs::create_dir_all(&sysroot_lib).expect(&format!("Failed to create {sysroot_lib:?}"));
+    let std_path = path_buf!(&sysroot_lib, "rustlib", target, "lib");
+    fs::create_dir_all(&std_path).expect(&format!("Failed to create {std_path:?}"));
 
-    //  Copy Kani libraries to inside sysroot folder.
-    let target_folder = Path::new(target_dir);
-    let macro_lib = format!("libkani_macros{}", lib_extension());
-    let kani_macros = path_buf!(target_folder, "debug", macro_lib);
-    cp(&kani_macros, &sysroot_lib).unwrap();
+    //  Copy Kani libraries to inside sysroot top folder.
+    copy_libs(&artifacts, &sysroot_lib, &is_kani_lib);
+    //  Copy Kani libraries to inside rustlib/<target>/lib/ folder.
+    copy_libs(&artifacts, &std_path, &is_std_lib);
+}
 
-    let kani_rlib_folder = path_buf!(target_folder, target, "debug");
-    cp_files(&kani_rlib_folder, &sysroot_lib, &is_rlib).unwrap();
+/// Is this artifact a library.
+fn is_rust_lib(artifact: &Artifact) -> bool {
+    artifact.target.kind.iter().any(|kind| match kind.as_str() {
+        "lib" | "rlib" | "proc-macro" => true,
+        "bin" | "dylib" | "cdylib" | "staticlib" | "custom-build" => false,
+        _ => unreachable!("Unknown crate type {kind}"),
+    })
+}
 
-    // Copy `std` libraries and dependencies to sysroot folder following expected path format.
-    let src_path = path_buf!(target_folder, target, "debug", "deps");
+/// Is this a kani library.
+fn is_kani_lib(artifact: &Artifact) -> bool {
+    is_rust_lib(artifact) && artifact.target.src_path.starts_with(env!("KANI_REPO_ROOT"))
+}
 
-    let dst_path = path_buf!(sysroot_lib, "rustlib", target, "lib");
-    fs::create_dir_all(&dst_path).unwrap();
-    cp_files(&src_path, &dst_path, &is_rlib).unwrap();
+/// Is this a std library or one of its dependencies.
+fn is_std_lib(artifact: &Artifact) -> bool {
+    is_rust_lib(artifact) && !is_kani_lib(artifact)
 }
```

Comment:
Non-blocking, but is there a more robust way to write these?

I also wonder if we could write unit tests purely as documentation (I'm less worried about actually testing behavior and more wondering "what files does this actually match?" It's very hard to understand just from code review.)